cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -94,23 +94,26 @@ public PushAppResponse push(PushAppOperationRequest pushRequest) throws NoChange
       ApplicationDetail app = pushRequest.getApp();
       writeAppDetail(repositoryManager, app);
 
-      List<PushAppResponse> responses = new ArrayList<>(commitAndPush(repositoryManager,
+      Collection<PushAppMeta> appMetas = commitAndPush(repositoryManager,
           pushRequest.getCommitDetails(),
           ImmutableList.of(new PushFile(
                   app.getName(),
                   app.getAppVersion(),
                   repositoryManager.getFileRelativePath(generateConfigFileName(app.getName()))
               )
           )
-      ));
-
-      // it should never be empty as in case of any error we will get an exception
-      return responses.get(0);
+      );
+      try {
+        String currentCommit = repositoryManager.getCurrentCommit();
```

Comment:
this is error prone, as it's possible the repository state has changed. It's better to return the commit hash as part of the commit method

Reply:
Yeah makes sense. I have changed the `commitAndPush` method to return a pair with commitid and pushmetas